### ISSUE TITLE
Support TOML/JSON, anything other than YML

### ISSUE THREAD

**Issue by MatthewCroughan** [F1bonacc1/process-compose#139]
It is not always easy to write YML files programatically, compared to something like JSON/TOML. It would be very nice to support something other than YML.

**Comment by thenonameguy** [comment]
JSON is always valid as YML (since YAML is a superset of JSON). Therefore it is already supported.
TOML support would be interesting though!

**Comment by srid** [comment]
This would help with avoiding IFD in `process-compose-flake`

https://github.com/Platonic-Systems/process-compose-flake/blob/b346fb30ad5fdee14e3edc3ba490eaa5db9ec0b7/nix/process-compose/settings/default.nix#L111-L113

**Comment by thenonameguy** [comment]
> This would help with avoiding IFD in `process-compose-flake`
> 
> 
> 
> https://github.com/Platonic-Systems/process-compose-flake/blob/b346fb30ad5fdee14e3edc3ba490eaa5db9ec0b7/nix/process-compose/settings/default.nix#L111-L113

Have you tried without the yq call? It should already work with the json file encoding with a .yml suffix.

**Comment by srid** [comment]
Yup, that works: https://github.com/Platonic-Systems/process-compose-flake/pull/59

I didn't know that YAML was considered a superset of JSON.
